### PR TITLE
Consolidate Graphic Elements to One Class

### LINKED ISSUE #30: Add Graphics Elements
Aside from the Box, none of the built in graphics element are implemented yet.

- [x] **Graphic circle** - PR #31, #42
^GC diameter, thickness, color 

- [x] **Graphic diagonal line** - PR #32, #42
^GD width, height, thickness, color, orientation	

- [X] **Graphic ellipse** #42
^GE width, height, thickness, color
	
- [x]  **Graphic symbol** #42
^GS orientation, height, width	

- [x] ~Also the box element is already included, but does not have the corner rounding or color attributes included.~ Fixed by PR #33 

- [x] ~Eventually it may be useful to add in a module for the related graphics values.~ Covered in #42

### PR DESCRIPTION
**Pull request by LagTag**
Ellipse, Box, Circle, Symbol, and Diagonal elements all added to one class. 

Added Spec for new graphic class. Deleted Diagonal and elipse classes (since these were only added a few weeks ago, I didn't deprecate them.

Deprecated warning for Box class as it is now redundant. Eventually will delete this class.

Closes #30

### CODE REVIEW

File: lib/zebra/zpl/graphic.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      def orientation=(value)
+        raise InvalidOrientationError unless value&.upcase.in?(["R","L"])
+        @orientation = value
+      end
+
+      def to_zpl
+        check_attributes
+        zpl = case graphic_type 
+        when "B"
+            "D#{graphic_width},#{graphic_height},#{line_thickness},#{color},#{orientation}"
```
Comment (mtking2) [inline lib/zebra/zpl/graphic.rb]: Does this need to be `B` instead of `D` here?

**Comment by mtking2** [comment]
Sweet. This is much cleaner 👌   Should close #30 now since everything is covered in this.

File: lib/zebra/zpl/graphic.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        @graphic_height = height
+      end
+
+      def color=(value)
+        raise InvalidColorError unless value&.upcase.in?(["B","W"])
+        @color = value
+      end
+
+      def orientation=(value)
+        raise InvalidOrientationError unless value&.upcase.in?(["R","L"])
```
Comment (mtking2) [inline lib/zebra/zpl/graphic.rb]: use `%w[R L].include?(value&.upcase)` instead of rails specific `.in?`

File: lib/zebra/zpl/graphic.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      def graphic_width=(width)
+        @graphic_width = width
+      end
+
+      def graphic_height=(height)
+        @graphic_height = height
+      end
+
+      def color=(value)
+        raise InvalidColorError unless value&.upcase.in?(["B","W"])
```
Comment (mtking2) [inline lib/zebra/zpl/graphic.rb]: use `%w[B W].include?(value&.upcase)` instead of rails specific `.in?`

File: lib/zebra/zpl/graphic.rb [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      DIAGONAL = "D"
+      CIRCLE   = "C"
+      SYMBOL   = "S"
+
+      def graphic_type=(type)
+        raise InvalidGraphicType unless %w(E B D C S).include? type
+        @graphic_type = type
+      end
+
+      def self.valid_graphic_type?(type)
```
Comment (LagTag) [inline lib/zebra/zpl/graphic.rb]: Guess I forgot to remove this method when I created the setter for graphic type

**Pull request merged**